readme_assets/system_diagram/system_architecture.xml

```diff
@@ -0,0 +1,133 @@
+          <mxGeometry x="525" y="590" width="120" height="40" as="geometry" />
+        </mxCell>
+        <mxCell id="JVIGxF9oKAkimninJbSp-16" value="frontend" style="rounded=0;whiteSpace=wrap;html=1;strokeWidth=1;fillColor=#f8cecc;strokeColor=#b85450;" parent="1" vertex="1">
+          <mxGeometry x="525" y="700" width="120" height="40" as="geometry" />
+        </mxCell>
+        <mxCell id="JVIGxF9oKAkimninJbSp-18" style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;startArrow=classic;startFill=1;exitX=0.5;exitY=0;exitDx=0;exitDy=0;entryX=0.5;entryY=1;entryDx=0;entryDy=0;entryPerimeter=0;" parent="1" source="JVIGxF9oKAkimninJbSp-16" target="JVIGxF9oKAkimninJbSp-15" edge="1">
+          <mxGeometry relative="1" as="geometry">
+            <mxPoint x="584.5" y="700" as="sourcePoint" />
+            <mxPoint x="585" y="660" as="targetPoint" />
+          </mxGeometry>
+        </mxCell>
+        <mxCell id="JVIGxF9oKAkimninJbSp-26" style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;entryX=0.5;entryY=0;entryDx=0;entryDy=0;startArrow=classic;startFill=1;endArrow=classic;endFill=1;" parent="1" source="JVIGxF9oKAkimninJbSp-21" target="JVIGxF9oKAkimninJbSp-10" edge="1">
+          <mxGeometry relative="1" as="geometry" />
+        </mxCell>
+        <mxCell id="JVIGxF9oKAkimninJbSp-21" value="Redis" style="strokeWidth=1;html=1;shape=mxgraph.flowchart.database;whiteSpace=wrap;fillColor=#dae8fc;strokeColor=#6c8ebf;" parent="1" vertex="1">
+          <mxGeometry x="520" y="250" width="130" height="70" as="geometry" />
+        </mxCell>
+        <mxCell id="JVIGxF9oKAkimninJbSp-58" value="-Search queries&lt;br&gt;-Tag Stats&amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp;&amp;nbsp;" style="text;html=1;align=left;verticalAlign=middle;resizable=0;points=[];autosize=1;strokeColor=none;fillColor=none;" parent="1" vertex="1">
+          <mxGeometry x="770" y="350" width="110" height="40" as="geometry" />
+        </mxCell>
+        <mxCell id="JVIGxF9oKAkimninJbSp-59" value="-Site cache&amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp;&amp;nbsp;&amp;nbsp; &amp;nbsp; &amp;nbsp;&amp;nbsp;&lt;br&gt;-Dead link cache" style="text;html=1;align=left;verticalAlign=middle;resizable=0;points=[];autosize=1;strokeColor=none;fillColor=none;" parent="1" vertex="1">
+          <mxGeometry x="590" y="360" width="125" height="41" as="geometry" />
+        </mxCell>
+        <mxCell id="JVIGxF9oKAkimninJbSp-60" value="-Detail views&lt;br&gt;-Lists&amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp;&lt;br&gt;-Content provider data&lt;br&gt;&lt;div style=&quot;&quot;&gt;&lt;span style=&quot;background-color: initial;&quot;&gt;-Link shortening data&lt;/span&gt;&lt;/div&gt;" style="text;html=1;align=left;verticalAlign=middle;resizable=0;points=[];autosize=1;strokeColor=none;fillColor=none;" parent="1" vertex="1">
```

Comment:
```suggestion
        <mxCell id="JVIGxF9oKAkimninJbSp-60" value="-Detail views&lt;br&gt;-Lists&amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp;&lt;br&gt;-Content provider data&lt;br&gt;&lt;div style=&quot;&quot;&gt;&lt;span style=&quot;background-color: initial;&quot;&gt;&lt;/span&gt;&lt;/div&gt;" style="text;html=1;align=left;verticalAlign=middle;resizable=0;points=[];autosize=1;strokeColor=none;fillColor=none;" parent="1" vertex="1">
```

Reply:
I'm not sure that this suggestion will work for PNG generation, but we should remove the `Link shortening data` here, too.